pkg/api/types.go

```diff
@@ -3023,6 +3023,10 @@ type NodeList struct {
 type NamespaceSpec struct {
 	// Finalizers is an opaque list of values that must be empty to permanently remove object from storage
 	Finalizers []FinalizerName
+
+	// NetworkPolicy describes the use of NetworkPolicy objects in this namespace. If not specified, then default policies will be applied and NetworkPolicy objects in the Namespace will be ignored
+	// +optional
+	NetworkPolicy *NamespaceNetworkPolicy
```

Comment:
NetworkPolicy is already namespace-scoped. But NetworkPolicies aren't restrictions, they're relaxations of a restriction. A namespace with 2 NetworkPolicies accepts more traffic than a a namespace with only 1 NetworkPolicy (generally speaking). So switching the namespace from "default allow" to "default deny" with the addition of the first NP is weird.

There was discussion on slack yesterday (https://kubernetes.slackarchive.io/sig-network/page-99/ts-1494875966879415) with the eventual decision that we need to wait for Tim to get back from his conference before making a decision, but one possibility was to leave Namespace unchanged and keep the annotation for now while we figure that out, but still move NetworkPolicy itself to networking/v1 for 1.7.